theurgy.lic

```diff
@@ -290,6 +290,10 @@ class Theurgy
 
   def study_wall
     return if DRStats.circle < 30
+    get_data('spells')
+      .spell_data
+      .select { |_name, properties| properties['cyclic'] }
+    return unless get_data('spells').spell_data.select { |name, _properties| DRSpells.active_spells.keys.include?(name) }.empty?
```

Comment:
Is this logic used in any other scripts? I'm wondering if there would be use in moving it to common-arcana as a helper.